chain/ethereum/src/block_ingestor.rs

```diff
@@ -84,29 +92,57 @@ where
             })
             .for_each(move |_| {
                 // Attempt to poll
-                static_self.do_poll().then(move |result| {
-                    if let Err(err) = result {
-                        // Some polls will fail due to transient issues
-                        match err {
-                            EthereumAdapterError::BlockUnavailable(_) => {
-                                trace!(
-                                    static_self.logger,
-                                    "Trying again after block polling failed: {}",
-                                    err
-                                );
+                static_self
+                    .do_poll()
+                    .then(move |result| {
+                        if let Err(err) = result {
+                            // Some polls will fail due to transient issues
+                            match err {
+                                EthereumAdapterError::BlockUnavailable(_) => {
+                                    trace!(
+                                        static_self.logger,
+                                        "Trying again after block polling failed: {}",
+                                        err
+                                    );
+                                }
+                                EthereumAdapterError::Unknown(inner_err) => {
+                                    warn!(
+                                        static_self.logger,
+                                        "Trying again after block polling failed: {}", inner_err
+                                    );
+                                }
                             }
-                            EthereumAdapterError::Unknown(inner_err) => {
-                                warn!(
+                        }
+
+                        // Continue polling even if polling failed
+                        future::ok(())
+                    })
+                    .inspect(move |_| {
+                        if *CLEANUP_BLOCKS {
+                            match static_self
+                                .chain_store
+                                .cleanup_cached_blocks(static_self.ancestor_count)
+                            {
```

Comment:
I liked the ability to control the interval. Do we think this is cheap enough to run every time?

Reply:
Ah, well, it's a development time thing, so let's not worry about it. 😄

Reply:
That, and I really wouldn't know what to tell users how to choose a good interval. Testing this locally with one syncing subgraph, it would generally remove one or two blocks, and that did not slow the block stream down at all, it's very quick. I had (from previous syncs) ~ 20k blocks in the DB; cleanup whittled it down to about 2k blocks while it was syncing without causing any noticeable delay.

On the 'continually download blocks': yeah, that phrasing could be improved, I just wanted to say something sufficiently scary so people don't turn this on in production settings. But I am open to rewording that, just not sure how.